### PR TITLE
replace custom build protocol with pubusb and create classes for diff…

### PR DESCRIPTION
**Pull request by GGAlanSmithee**
@brokenprogrammer 

another small improvement, please merge #13 first though

### CODE REVIEW

File: src/peers/builder-peer.js [hunk 1]
```diff
@@ -31,11 +31,8 @@ class BuilderPeer {
 
     await peer.#node.start()
 
-    peer.#node.pubsub.on(request_build_topic, (msg) => {
-      logger.logBuilderSubMessage(
-        msg.from,
-        stringFromUInt8Array(msg.data) // TODO(Alan): JSON parse proper metadata
-      )
+    peer.#node.pubsub.on(request_build_topic, ({ from, data }) => {
```
Comment (brokenprogrammer) [inline src/peers/builder-peer.js:36]: What does this `.#` syntax mean?

**Review by brokenprogrammer — APPROVED** [review]
LGTM! I can't merge anything in this repo so I'm just reviewing.

File: src/peers/builder-peer.js (hunk 1 above)
Comment (GGAlanSmithee) [reply src/peers/builder-peer.js:36]: It's ecmascript class privat fields: https://github.com/tc39/proposal-class-fields#private-fields

**Comment by GGAlanSmithee** [comment]
> LGTM! I can't merge anything in this repo so I'm just reviewing.

Thanks, we will have to fix that - did not know

**Comment by GGAlanSmithee** [comment]
I rebased it on main and added back a missing log, merging

**Pull request merged**